### PR TITLE
fix: Created Product model to encompass Yarn model and future models.

### PR DESCRIPTION
**Pull request by DariBerrie**
Hi guys! 
I had a think about our Yarn model and realized that we have pigeon-holed ourselves into ONLY having users create yarn listings though they may be interested in selling other types of things like tools, buttons, etc.

So I created a parent Product model that will be using [**Single Table Inheritance**](https://mattwal.medium.com/sti-polymorphism-and-abstract-classes-rails-a7f588668cbb#:~:text=Single%20Table%20Inheritance%20(STI) to allow the creation of other models like Yarn, Needles, Crochets Hooks, etc. Interestingly enough, adding differing attributes to our children classes doesn't seem to be an easy thing to achieve in Rails so out of all the inheritance options proposed, STI seemed best. This means that some of our children classes may have some **nil** values (like :size for Yarn), but I think it's an OK tradeoff and an easier implementation. 

This also means we can remove the Yarn migrations and allow Yarn to inherit from the Product class (visible in "yarn.rb"). And you can also see that this allows us to still seed using the Yarn class w/o making any substantial changes.

I will also create a Products Controller and routes to coincide with our new model. For right now, I will keep the existing controllers as they currently are, and we can make updates as needed. 

Let me know what your thoughts are!

### CODE REVIEW

**Comment by zynolia** [comment]
Thank you for working on this Dari, i think it makes sense to have a product parent model

**Review by zynolia — APPROVED** [review]

File: app/models/order.rb [hunk 1]
```diff
@@ -1,4 +1,4 @@
 class Order < ApplicationRecord
   belongs_to :user
-  belongs_to :yarn
```
Comment (AmandaPerino) [inline app/models/order.rb:3]: Question: Would the association not be an Order "has_many :products"?

Comment (DariBerrie) [reply app/models/order.rb:3]: I agree!

Comment (AmandaPerino) [reply app/models/order.rb:3]: Okay!

**Review by AmandaPerino — APPROVED** [review]
All looks good if we are keeping the 1-to-1 association between Orders and Products.

**Pull request merged**